### PR TITLE
Rename View() to Page()

### PR DESCRIPTION
**Pull request by rynowak**
This requires us to introduce another base class between RazorPageBase and
Page - you need this because you aren't allow to have Page.Page().

### CODE REVIEW

**Comment by rynowak** [comment]
I accidentally made the PR based on dev, but this is actually for preview1.

File: src/Microsoft.AspNetCore.Mvc.RazorPages/PageModel.cs [hunk 1]
```diff
@@ -799,6 +794,12 @@ public virtual NotFoundObjectResult NotFound(object value)
             => new NotFoundObjectResult(value);
 
         /// <summary>
+        /// Creates a <see cref="PageResult"/> object that renders the page.
+        /// </summary>
+        /// <returns>The <see cref="PageResult"/>.</returns>
+        public virtual PageResult Page() => new PageResult(PageContext.Page, this);
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/PageModel.cs]: Another alternative might be to rename this method to `CurrentPage()`. It seems a bit clearer too in the intent of the method?

Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/PageModel.cs]: nah. We'd rather deal with base-class grossness

**Comment by rynowak** [comment]
FYI I double checked this and it looks like there's no dependency in the Razor repo on this. None of the tests call this method.

**Comment by rynowak** [comment]
b1b3a816cc808037d3e6bb45c955b344792a9343 in 2.0.0-preview1
0c70cdb8dfee85043c84a3656c5424992643406f merged into dev

**Pull request closed without merging**